### ISSUE TITLE
tlc with NixOS

### ISSUE THREAD

**Issue by err0r500** [alygin/vscode-tlaplus#180]
Hello,
If I try to use the extension installed with `vscode-utils.extensionsFromVscodeMarketplace` (in `configuration.nix`), checking the model generates files in the `./states` folder but they get deleted instantly once the check is finished. 

Not sure if there are lots of NixOS users here, but do you have an idea what could cause this ?
The rest works fine (sany, tla2tex) ; also, tlc works fine in the terminal

**Comment by lemmy** [comment]
What does this have to do with NixOS and why would you want to keep TLC's temporary data?

**Comment by err0r500** [comment]
I was about to edit the question, the problem is that no result is displayed (and I thought it came from the deleted files in states).

Here's the NixOS related problem : I've got 2 ways to install vscode extensions : globally or per user (I can't mix these 2 install modes). 
The problem I describe happens only if I install the extension globally, everything works fine if I install it as a user.
So why don't I just keep it like that ? Because liveshare only works if installed globally...

**Comment by lemmy** [comment]
Sounds to me as if `states` is a red herring -- TLC deletes temp files in `states` after execution.

**Comment by err0r500** [comment]
The main difference I know between the 2 install modes is that the extension install dir is read-only with global install. 
I didn't see write while quickly browsing the extension code, but maybe I missed something ?

**Comment by lemmy** [comment]
TLC seems to succeed to create `states` itself, no?

**Comment by err0r500** [comment]
yeap, but `states` is in the spec folder, I meant the installation dir of the extension itself (`~/.vscode/extensions/<ext-name>` in most distros)

**Comment by kubukoz** [comment]
@err0r500 you might actually be having the same problem I was having - check out #196.

```nix
tla = pkgs.vscode-utils.buildVscodeExtension rec {
# I had to run `npm install` in there first - less than ideal, but it works...
  src = ../../IdeaProjects/vscode-tlaplus;
  name = "vscode-tlaplus";
  publisher = "alygin";
  vscodeExtUniqueId = "${publisher}.${name}";
  version = "1.5.2";
}
```

**Comment by alygin** [comment]
The fix is released in [v1.5.3](https://github.com/alygin/vscode-tlaplus/releases/tag/v1.5.3).
@err0r500, feel free to reopen the issue if the extension still doesn't work on NixOS.
